src/routes/AppRoutes.tsx

```diff
@@ -15,24 +16,40 @@ import useAuth from '@/modules/auth/hooks/useAuth';
 import useSessionStatus from '@/modules/auth/hooks/useSessionStatus';
 import SystemStatus from '@/modules/systemStatus/components/SystemStatus';
 
-export interface RouteLocationState {
-  /** Previous pathname, so we can redirect to it when logging back in */
-  prev?: string;
-  /** Whether to clear/ignore the previous pathname. For example when someone clicks "Sign Out," the previous location shouldn't be maintained when they log back
-  in.  */
-  clearPrev?: boolean;
+const REQUESTED_PATH_KEY = 'hmis_requested_path';
+
+// allow-list for urls that can be redirected to after sign-in
+function pathForRedirect(arg: string): string | undefined {
+  const path = arg.trim();
+  const idPattern = '[a-zA-Z0-9=]+';
+  const patterns = [
+    `^/client/${idPattern}/.*`,
+    `^/projects/${idPattern}/.*`,
+    `^/projects`,
+    `^/admin/.*`,
+  ];
```

Comment:
why not support redirecting to any url? this seems ok, just wondering.

Reply:
oh and we definitely want to support redirecting to a search route, e.g. `/?textSearch=foo`

Reply:
Oh yeah, seems like we shouldn't expose search terms in the url but that's probably a different issue.

There may be certain paths we might want to prevent redirecting into after login, for example a page nested in a deep workflow or a page that might trigger a side-effect action (maybe something we build in the future).

But maybe it's not worth the bother here. To be less trouble, I can invert this into a deny list and by default we allow redirects to any path after login.

Reply:
Ah okay, I see. We don't have pages like that right now, as far as I know, but we may in the future. I'm happy to leave it either way